### PR TITLE
Bug 1911269: hide "waiting for builds" info message when a build is present

### PR DESCRIPTION
**Pull request by debsmita1**
**Fixes:**
https://issues.redhat.com/browse/ODC-5069

**Analysis / Root cause:**
It is not being checked if a successful build exists

**Solution Description:**
- Checking if there exist some pod(s) having at least one(or more) container(s) with image id
- added translation

**GIF:**
build-info-msg

**Browser conformance:**
- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/public/components/overview/pods-overview.tsx [hunk 1]
```diff
@@ -135,15 +135,15 @@ export const PodsOverviewContent: React.SFC<PodsOverviewContentProps> = ({
...
+  isBuildComplete,
 }) => {
   const {
     metadata: { name, namespace },
   } = obj;
 
   const [showWaitingPods, setShowWaitingPods] = React.useState(false);
   const showWaitingForBuildAlert =
-    hasBuildConfig && isDeploymentGeneratedByWebConsole(obj) && pods.some(isPodWithoutImageId);
+    !isBuildComplete && isDeploymentGeneratedByWebConsole(obj) && pods.some(isPodWithoutImageId);
```
Comment (divyanshiGupta) [inline frontend/public/components/overview/pods-overview.tsx:146]: This doesn't seem like the right solution as when the build completes this `pods.some(isPodWithoutImageId)` should return `false`. I think you should look into fixing this check and see why this isn't returning `false` when the build completes.

File: frontend/public/components/overview/pods-overview.tsx [hunk 2]
```diff
@@ -68,6 +68,9 @@ const isPodWithoutImageId = (pod: PodKind) =>
   pod.status?.phase === 'Pending' &&
   pod.status?.containerStatuses?.some((containerStatus) => !containerStatus.imageID);
 
+const isPodWithImageId = (pod: PodKind) =>
```
Comment (rohitkrai03) [inline frontend/public/components/overview/pods-overview.tsx:71]: Why do we need two different checks here `isPodWithoutImageId` and `isPodWithImageId `? Can we not use just one check for both cases?

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: frontend/public/components/overview/pods-overview.tsx (hunk 2 above)
Comment (divyanshiGupta) [reply frontend/public/components/overview/pods-overview.tsx:71]: ^^ @debsmita1

Comment (debsmita1) [reply frontend/public/components/overview/pods-overview.tsx:68]: removed the usage of `isPodWithoutImageId`

File: frontend/public/components/overview/pods-overview.tsx [hunk 3]
```diff
@@ -140,44 +140,47 @@ export const PodsOverviewContent: React.SFC<PodsOverviewContentProps> = ({
...
           isInline
           variant="info"
-          title="Waiting for the build"
+          title={t('public~Waiting for the build')}
           actionLinks={
             <AlertActionLink onClick={() => setShowWaitingPods(!showWaitingPods)}>
-              {`${showWaitingPods ? 'Hide' : 'Show'} waiting pods with errors`}
+              {t('public~{{waitingPods}} waiting pods with errors', {
+                waitingPods: showWaitingPods ? 'Hide' : 'Show',
+              })}
```
Comment (jerolimov) [inline frontend/public/components/overview/pods-overview.tsx:177]: Hi, this doesn't translate Hide and Show, right? So my suggest is:

```suggestion
              {showWaitingPods ? t('public~Hide waiting pods with errors') : t('public~Show waiting pods with errors')}
```

Comment (jerolimov) [reply frontend/public/components/overview/pods-overview.tsx:177]: I notice that your translation works fine, but its maybe still easier to translate two separate messages.

Comment (debsmita1) [reply frontend/public/components/overview/pods-overview.tsx]: Done

**Review by jerolimov — APPROVED** [review]
Tested this locally. Works fine for the initial build and new builds. i18n looks good.

/lgtm

**Comment by divyanshiGupta** [comment]
@debsmita1 I see the following issues:
Peek 2021-01-13 17-07

`Build Alert` is shown first and then the pods every time on going to the resources tab. Ideally we should see the loading component and then the pods like it used to be before.

Peek 2021-01-13 17-05

On scaling down to 0 pods `Buid Alert` is shown. Ideally only the `No pod resources found` empty state should be shown.

**Comment by debsmita1** [comment]
> @debsmita1 I see the following issues:
> Peek 2021-01-13 17-07
> 
> `Build Alert` is shown first and then the pods every time on going to the resources tab. Ideally we should see the loading component and then the pods like it used to be before.
> 
> Peek 2021-01-13 17-05
> 
> On scaling down to 0 pods `Buid Alert` is shown. Ideally only the `No pod resources found` empty state should be shown like before.

Fixed !

File: frontend/public/components/overview/pods-overview.tsx [hunk 4]
```diff
@@ -128,22 +130,25 @@ const PodsOverviewList: React.SFC<PodOverviewListProps> = ({ pods }) => (
...
 }) => {
   const {
     metadata: { name, namespace },
   } = obj;
-
+  const { t } = useTranslation();
   const [showWaitingPods, setShowWaitingPods] = React.useState(false);
   const showWaitingForBuildAlert =
-    hasBuildConfig && isDeploymentGeneratedByWebConsole(obj) && pods.some(isPodWithoutImageId);
+    !buildConfigData?.buildConfigs?.[0].builds.some((build) => isComplete(build)) &&
```
Comment (rohitkrai03) [inline frontend/public/components/overview/pods-overview.tsx:150]: What if there are no buildConfigs in the returns object?

File: frontend/public/components/overview/pods-overview.tsx [hunk 5]
```diff
@@ -154,32 +159,38 @@ export const PodsOverviewContent: React.SFC<PodsOverviewContentProps> = ({
...
+              ? t('public~Hide waiting pods with errors')
+              : t('public~Show waiting pods with errors')}
+          </AlertActionLink>
+        }
+      >
+        {t(
+          'public~Waiting for the first build to run successfully. You may temporarily see "ImagePullBackOff" and "ErrImagePull" errors while waiting.',
+        )}
+      </Alert>
+    ) : null;
```
Comment (rohitkrai03) [inline frontend/public/components/overview/pods-overview.tsx:182]: No need to create a function here, you can just write jsx directly.
```suggestion
  const PodAlert = showWaitingForBuildAlert ? (
      <Alert
        isInline
        variant="info"
        title={t('public~Waiting for the build')}
        actionLinks={
          <AlertActionLink onClick={() => setShowWaitingPods(!showWaitingPods)}>
            {showWaitingPods
              ? t('public~Hide waiting pods with errors')
              : t('public~Show waiting pods with errors')}
          </AlertActionLink>
        }
      >
        {t(
          'public~Waiting for the first build to run successfully. You may temporarily see "ImagePullBackOff" and "ErrImagePull" errors while waiting.',
        )}
      </Alert>
    ) : null;
```

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: frontend/public/components/overview/pods-overview.tsx (hunk 4 above)
Comment (debsmita1) [reply frontend/public/components/overview/pods-overview.tsx]: added check for buildConfigs

**Review by rohitkrai03 — APPROVED** [review]
/approve

**Review by divyanshiGupta — APPROVED** [review]
/lgtm

File: frontend/public/components/overview/pods-overview.tsx [hunk 6]
```diff
@@ -154,32 +160,37 @@ export const PodsOverviewContent: React.SFC<PodsOverviewContentProps> = ({
   const errorPodCount = _.size(_.filter(pods, (pod) => isPodError(pod)));
   const podsShown = Math.max(Math.min(errorPodCount, MAX_ERROR_PODS), MAX_PODS);
   const linkTo = allPodsLink || `${resourcePath(referenceFor(obj), name, namespace)}/pods`;
-  const emptyMessage = emptyText || 'No Pods found for this resource.';
+  const emptyMessage = emptyText || t('public~No Pods found for this resource.');
+
+  const PodAlert = showWaitingForBuildAlert ? (
```
Comment (jerolimov) [inline frontend/public/components/overview/pods-overview.tsx:165]: Sorry, small nit

```suggestion
  const podAlert = showWaitingForBuildAlert ? (
```

File: frontend/public/components/overview/pods-overview.tsx [hunk 7]
```diff
@@ -154,32 +160,37 @@ export const PodsOverviewContent: React.SFC<PodsOverviewContentProps> = ({
...
-            <AlertActionLink onClick={() => setShowWaitingPods(!showWaitingPods)}>
-              {`${showWaitingPods ? 'Hide' : 'Show'} waiting pods with errors`}
-            </AlertActionLink>
-          }
-        >
-          Waiting for the first build to run successfully. You may temporarily see
-          "ImagePullBackOff" and "ErrImagePull" errors while waiting.
-        </Alert>
-      ) : null}
+      {buildConfigData.loaded && !buildConfigData.loadError && PodAlert}
```
Comment (jerolimov) [inline frontend/public/components/overview/pods-overview.tsx:193]:
```suggestion
      {buildConfigData.loaded && !buildConfigData.loadError && podAlert}
```

File: frontend/public/locales/en/public.json [hunk 8]
```diff
@@ -384,6 +384,13 @@
   "Limit Ranges": "Limit Ranges",
   "Chargeback": "Chargeback",
   "Custom Resource Definitions": "Custom Resource Definitions",
+  "View logs": "View logs",
+  "No Pods found for this resource.": "No Pods found for this resource.",
+  "Waiting for the build": "Waiting for the build",
+  "Hide waiting pods with errors": "Hide waiting pods with errors",
+  "Show waiting pods with errors": "Show waiting pods with errors",
+  "Waiting for the first build to run successfully. You may temporarily see \"ImagePullBackOff\" and \"ErrImagePull\" errors while waiting.": "Waiting for the first build to run successfully. You may temporarily see \"ImagePullBackOff\" and \"ErrImagePull\" errors while waiting.",
+  "View all {{podSize}}": "View all {{podSize}}",
```
Comment (jerolimov) [inline frontend/public/locales/en/public.json:393]: :+1:

**Review by jerolimov — APPROVED** [review]
Thanks for this quick fix of this nit

/lgtm

**Pull request merged**